go/logictest/runner.go

```diff
@@ -238,19 +272,67 @@ func runTestFile(harness Harness, file string) {
 		panic(err)
 	}
 
+	dnr := false
+	ll := newLoggingLock()
```

Comment:
You need a new lock for every record. Reusing it like this will introduce more race conditions and buggy behavior